backend/ShootTheMoon/ShootTheMoon/Network/ShootServer.cs

```diff
@@ -474,5 +518,71 @@ public override async Task<StatusResponse> TakeSeat(TakeSeatRequest request, Ser
 
             return r;
         }
+
+        public override async Task<StatusResponse> CreateBid(Proto.Bid request, ServerCallContext context) {
+            string uuid = context.RequestHeaders.GetValue(GAME_ID);
+            string clientToken = context.RequestHeaders.GetValue(CLIENT_TOKEN);
+
+            StatusResponse r = new StatusResponse();
+            Game.Game game;
+
+            try {
+                game = games[uuid];
+            }
+            catch (KeyNotFoundException) {
+                r.Success = false;
+                r.ErrorNum = (int)ErrorCode.GAME_NOT_FOUND;
+                r.ErrorText = "Game Not Found";
+                return r;
+            }
+
+            try {
+                RpcClient client = FindClient(game, clientToken);
+                Log.Debug("Received a bid of (tricks: {0}, suit: {1}, shoot: {2}) by {3}", request.Tricks, request.Trump, request.ShootNum, client.Name);
+                
+                Game.Trump trump = null;
+                foreach (KeyValuePair<Game.Trump, Proto.Trump> t in GameTrumpToProtoTrump) {
+                    if (t.Value == request.Trump) {
+                        trump = t.Key;
+                        break;
+                    }
+                }
+                game.MakeBid(request.Tricks, trump, request.ShootNum, false, client);
+            }
+            catch (KeyNotFoundException) {
+                r.Success = false;
+                r.ErrorNum = (int)ErrorCode.CLIENT_NOT_FOUND;
+                r.ErrorText = "Client Not Found";
+                return r;
+            }
+
+            r.Success = true;
+            r.ErrorNum = (int)ErrorCode.SUCCESS;
+            r.ErrorText = "";
+
+            return r;
+        }
+
+        /*
```

Comment:
do we need these?

Reply:
Eventually, and this was me being lazy and bringing the function prototypes in now so I don't have to go looking for them later.